src/imitation/algorithms/preference_comparisons.py

```diff
@@ -889,15 +892,33 @@ def _make_data_loader(self, dataset: PreferenceDataset) -> data_th.DataLoader:
 
     def _train(self, dataset: PreferenceDataset, epoch_multiplier: float = 1.0) -> None:
         """Trains for `epoch_multiplier * self.epochs` epochs over `dataset`."""
-        dataloader = self._make_data_loader(dataset)
+        if self.regularizer:
+            val_length = int(len(dataset) * self.val_split)
+            train_length = len(dataset) - val_length
+            train_dataset, val_dataset = data_th.random_split(dataset, lengths=[train_length, val_length])
```

Comment:
Support seeding (via `generator`)?

Reply:
Thanks, yes. This was in the original version, but got over-written when I pulled the changes from `reward_ensemble`.